docs/modules/reference/pages/service-assurance/monitors/PtpMonitor.adoc

```diff
@@ -0,0 +1,60 @@
+
+= PtpMonitor
+:description: Learn how to configure and use the PtpMonitor in OpenNMS {page-component-title} to monitor the correct PTP port state for an interface.
+
+The Precision Time Protocol allows the precise synchronization of clocks in measurement and control systems implemented with packet-based networks.
+OpenNMS can monitor the port state of SNMP-enabled interfaces by using the PTP MIB defined in RFC 8173.
+During the poll, it queries the device for the available clock-ports and their states and compares with a given expected state.
+If the state equals the desired state the service is considered up and down otherwise.
+
+== Monitor facts
+
+[cols="1,7"]
+|===
+| Class Name
+| `org.opennms.netmgt.poller.monitors.PtpMonitor`
+|===
+
+== Configuration and Use
+
+.Optional monitor-specific parameters for the PtpMonitor
+[options="header"]
+[cols="1,3,2"]
+|===
+| Parameter
+| Description
+| Default
+
+| clock-port
+| Name of the interface that must be in the specified state.
+| n/a
+
+| port-state
+| The desired state that the port must be in.
+| n/a
+
+|===
+
+This monitor implements the <<reference:service-assurance/introduction.adoc#ref-service-assurance-monitors-common-parameters, Common Configuration Parameters>>.
+
+== Examples
+
+Example uses CentOS/RHEL path name.
+For Debian/Ubuntu, use `/var/lib/opennms/rrd/response`.
+
+Note that you must include the `monitor` section for each service in your definition.
+
+[source, xml]
+----
+ <service name="PTP" interval="300000" user-defined="false" status="on">
+     <pattern><![CDATA[^PTP-(?<clockPort>.*)-(?<portState>.*)$]]></pattern>
+     <parameter key="rrd-repository" value="${install.share.dir}/rrd/response" />
+     <parameter key="rrd-base-name" value="ptp" />
```

Comment:
We can probably eliminate the problem with ${install.share.dir} removing this configuration parameter. I think there is a default set in the poller we can rely on.

We should probably also remove the configuration:

```
<parameter key="rrd-base-name" value="ptp" />
```

If you use the pattern match and have multiple services defined, they will write all in the same ptp RRD file.

Reply:
done.

Reply:
> We can probably eliminate the problem with ${install.share.dir} removing this configuration parameter. I think there is a default set in the poller we can rely on.

Can we remove this one here as well and rely on the default the monitor code which will always match the install.dir anyway?

Reply:
> We can probably eliminate the problem with ${install.share.dir} removing this configuration parameter. I think there is a default set in the poller we can rely on.

@christianpape Can we remove this one here as well and rely on the default monitor code which will always match the ${install.share.dir} anyway?

Reply:
> We can probably eliminate the problem with ${install.share.dir} removing this configuration parameter. I think there is a default set in the poller we can rely on.

@christianpape Can we remove this one here as well and rely on the default monitor code which will always match the ${install.share.dir} anyway?

Reply:
done.